### PR TITLE
VACMS-3197: Add pathauto patterns.

### PR DESCRIPTION
**Pull request by ethanteague**
## Description
See #3197 
This pr adds pathauto patterns for learning center types and associated terms.
This will be the first of two.

## Testing done
Behat dst.

## Screenshots
N/A

## QA steps
**As and admin:**
- Go to `/admin/config/search/path/patterns/learning_center` and visually verify that the following items are selected: `Checklist, Step-by-step, Media list- Images, Media list - Videos, Q&A, FAQ - multiple Q&As, Learning Center Detail page`
- Go to `/admin/config/search/path/patterns/landing_page` and verify that `Landing page` is selected
- Go to `/admin/config/search/path/patterns/learning_center_category` and verify that `Learning Center Categories` is selected
- Go to `/admin/config/search/path/patterns/taxonomy_terms` and verify that `Topics, Audience - Beneficiaries, Audience - non-beneficiaries` are selected

### CODE REVIEW

**Comment by swirtSJW** [comment]
@ethanteague  I think we need some QA steps here that involve saving some nodes with certain values and validating the paths that get created.   

Ideally it would be nice to have a learning-center-content.feature that tests these paths.  Then we can use it for other tests too as we expand that product.

**Comment by ethanteague** [comment]
> @ethanteague I think we need some QA steps here that involve saving some nodes with certain values and validating the paths that get created.
> 
> Ideally it would be nice to have a learning-center-content.feature that tests these paths. Then we can use it for other tests too as we expand that product.

Added some tests, @swirtSJW :+1:

File: config/sync/pathauto.pattern.learning_center.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+uuid: ac26c7ac-f5a5-424e-83bc-06babfaac01c
+langcode: en
+status: true
+dependencies:
+  module:
+    - node
+id: learning_center
+label: 'Learning Center'
```
Comment (oksana-c) [inline config/sync/pathauto.pattern.learning_center.yml:8]:
```suggestion
id: resources_support
label: 'Resources and support'
```

File: config/sync/pathauto.pattern.learning_center_category.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+uuid: 9672dbe2-9354-4708-a7a3-267c34b799ed
+langcode: en
+status: true
+dependencies:
+  module:
+    - ctools
+    - taxonomy
+id: learning_center_category
+label: 'Learning Center Category'
```
Comment (oksana-c) [inline config/sync/pathauto.pattern.learning_center_category.yml:9]:
```suggestion
id: resources_support_category
label: 'Resources and support category terms'
```

File: config/sync/pathauto.pattern.taxonomy_terms.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+uuid: 47be66c7-ece7-45e1-a63c-5c92ee31d9b3
+langcode: en
+status: true
+dependencies:
+  module:
+    - ctools
+    - taxonomy
+id: taxonomy_terms
+label: 'Taxonomy terms'
```
Comment (oksana-c) [inline config/sync/pathauto.pattern.taxonomy_terms.yml:9]:
```suggestion
id: topics_audience_terms
label: 'Topics and Audience taxonomy terms'
```

**Review by oksana-c — COMMENTED** [review]
added minor naming recommendations

are we adding any specific logic to lock the alias after publishing in this PR?

File: tests/behat/Drupal/FeatureContext.php [hunk 4]
```diff
@@ -363,4 +363,23 @@ public function thenSelectorShouldNotContain($selector, $item) {
...
+   * @throws Exception
+   *
+   * @Then current path alias should be :path
+   */
+  public function theCurrentPathAliasShouldBe($path) {
+    global $base_url;
+
+    $url = $this->getSession()->getCurrentUrl();
+    if ($url !== $base_url . $path) {
+      throw new \Exception('The page url ' . $url . ' does not match test enry path ' . $base_url . $path);
```
Comment (swirtSJW) [inline tests/behat/Drupal/FeatureContext.php:381]: Is there a reason we could not make use of  the existing step `And the url should match "/blah/blah"`

https://github.com/department-of-veterans-affairs/va.gov-cms/blob/8046157610705dfe0e910e7b9fbf9133b542d421/tests/behat/features/decoupled.feature#L43

**Review by swirtSJW — COMMENTED** [review]
Just the one question

File: tests/behat/Drupal/FeatureContext.php (hunk 4 above)
Comment (ethanteague) [reply tests/behat/Drupal/FeatureContext.php]: > Is there a reason we could not make use of the existing step `And the url should match "/blah/blah"`
> 
> https://github.com/department-of-veterans-affairs/va.gov-cms/blob/8046157610705dfe0e910e7b9fbf9133b542d421/tests/behat/features/decoupled.feature#L43

D'oh!!! I was working harder not smarter here, @swirtSJW - updated :+1:

**Review by swirtSJW — APPROVED** [review]
Looks great Ethan thank you.

**Comment by ethanteague** [comment]
> Looks great Ethan thank you.

Thanks for catching that duplicate step logic, @swirtSJW :+1:

**Pull request merged**